modules/libcom/src/calc/calcPerform.c

```diff
@@ -46,6 +46,7 @@ LIBCOM_API long
     calcPerform(double *parg, double *presult, const char *pinst)
 {
     double stack[CALCPERFORM_STACK+1];  /* zero'th entry not used */
+    memset(stack, 0, sizeof(stack));
```

Comment:
FYI the Appveyor build failures from the VS2012 compiler for this PR were due to your `memset()` calls preceding some variable declarations — all our C code still puts all declarations before statements inside a block, which was required by older C standards.

Reply:
Appreciate you taking the time to look over this!

You're definitely right- this is not the correct solution, I should've dug into the code a bit more. 

I believe that adding a stack position check before dereferencing `pstacktop->name` would fix these issues. With the following change, all calc tests pass, and valgrind no longer complains about uninitialized memory use when parsing `0)`
```diff
diff --git a/modules/libcom/src/calc/postfix.c b/modules/libcom/src/calc/postfix.c
index 7457abaab..d9dce515d 100644
--- a/modules/libcom/src/calc/postfix.c
+++ b/modules/libcom/src/calc/postfix.c
@@ -335,6 +335,10 @@ LIBCOM_API long
             break;
 
         case SEPERATOR:
+            if (pstacktop == stack) {
+                *perror = CALC_ERR_BAD_SEPERATOR;
+                goto bad;
+            }
             /* Move operators to the output until open paren */
             while (pstacktop->name[0] != '(') {
                 if (pstacktop <= stack+1) {
@@ -353,6 +357,10 @@ LIBCOM_API long
             break;
 
         case CLOSE_PAREN:
+            if (pstacktop == stack) {
+                *perror = CALC_ERR_PAREN_NOT_OPEN;
+                goto bad;
+            }
             /* Move operators to the output until matching paren */
             while (pstacktop->name[0] != '(') {
                 if (pstacktop <= stack+1) {
```

For now I've dropped the problematic commit from this PR. If this fix looks acceptable, I can push it.

Reply:
Those changes do seem to work, if you want to add them back would you also add tests for `0)` and `0,` to the modules/libcom/test/epicsCalcTest.cpp file, there are a bunch of calls to `testBadExpr()` towards the end of the `MAIN()` routine and these belong there. You would also need to change the number of tests in the call to `testPlan()`.

Thanks!

Reply:
Alright, added in the latest diff!

Reply:
Thanks, looks fine by me now. The Core Developers group will probably take a look at this sometime, we're a bit behind in approving PRs at the moment though so I can't predict when it will actually get merged.